crates/astria-sequencer/src/mempool.rs

```diff
@@ -114,6 +128,74 @@ impl std::hash::Hash for EnqueuedTransaction {
     }
 }
 
+const TX_TTL: i64 = 600; // 10 minutes 
+const INVALID_CACHE_SIZE: usize = 4096;
+const TX_INVALID_TTL: i64 = 60; // 60 seconds
+
+/// `InvalidCache` provides for signaling to `CometBFT` what transactions
+///  can be removed from the `CometBFT` mempool.
+///
+/// The cache places a TTL on the entries, allowing users
+/// to request their transaction to be processed again after the
+/// ttl has expired. This is useful for users who want to re-run a transaction
+/// that failed to execute or to re-add a transaction that has expired.
+#[derive(Clone)]
+pub(crate) struct InvalidCache {
+    cache: HashSet<[u8; 32]>,
+    time_added: HashMap<[u8; 32], i64>,
+    remove_queue: VecDeque<[u8; 32]>,
+    max_size: usize,
+    time_to_live: i64,
+    size: usize,
+}
+
+impl InvalidCache {
+    fn new(max_size: usize, time_to_live: i64) -> Self {
+        Self {
+            cache: HashSet::new(),
+            time_added: HashMap::new(),
+            remove_queue: VecDeque::with_capacity(max_size),
+            max_size,
+            time_to_live,
+            size: 0,
+        }
+    }
+
+    fn cached(&self, tx_hash: [u8; 32]) -> bool {
```

Comment:
this funtion returns `true` if the tx is invalidated but not expired right? could you add a comment for that or maybe change the function name to `cached_and_not_expired` for clarity?

Reply:
is this the right behaviour? based on the logic for adding txs to the invalid cache and when to remove them from cometbft, should this return true if the tx is expired?

Reply:
the comment was ambiguous, it was returning true if it was expired in the app's mempool but not expired in the cache. refactored to make it more clear